sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasPermissions.cs

```diff
@@ -3,56 +3,94 @@
 
 using System;
 using System.ComponentModel;
+using System.Runtime.CompilerServices;
 using System.Text;
+using Azure.Storage.Sas;
 
 namespace Azure.Storage.Sas
 {
     /// <summary>
     /// <see cref="BlobSasPermissions"/> supports reading and writing
-    /// permissions string for a blob's access policy.  Use <see cref="ToString"/>
+    /// permissions string for a blob's access policy.  Use ToString
     /// to generate a permissions string you can provide to
     /// <see cref="BlobSasBuilder.Permissions"/>.
     /// </summary>
-    public struct BlobSasPermissions : IEquatable<BlobSasPermissions>
+    [Flags]
+    public enum BlobSasPermissions
     {
+        /// <summary>
+        ///
+        /// </summary>
+        None = 0,
+
         /// <summary>
         /// Get or sets whether Read is permitted.
         /// </summary>
-        public bool Read { get; set; }
+        Read = 1,
 
         /// <summary>
         /// Get or sets whether Add is permitted.
         /// </summary>
-        public bool Add { get; set; }
+        Add = 2,
 
         /// <summary>
         /// Get or sets whether Create is permitted.
         /// </summary>
-        public bool Create { get; set; }
+        Create = 4,
 
         /// <summary>
         /// Get or sets whether Write is permitted.
         /// </summary>
-        public bool Write { get; set; }
+        Write = 8,
 
         /// <summary>
         /// Get or sets whether Delete is permitted.
         /// </summary>
-        public bool Delete { get; set; }
+        Delete = 16,
+
+        /// <summary>
+        ///
+        /// </summary>
+        All = ~None
```

Comment:
Tip: recommend adding a trailing comma, which makes future diffs (though, in this case, I doubt anything would ever come after this) easier.